### PR TITLE
Add ASR to vacancy ATS integrations

### PR DESCRIPTION
**Pull request by ddippolito**
## Trello card URL
https://trello.com/c/0guG3Zoz/858-adding-all-school-roles-to-ats-integrations

## Changes in this PR:

Allow for multiple roles in ATS integrations

### CODE REVIEW

File: app/vacancy_sources/vacancy_source/source/ark.rb [hunk 1]
```diff
@@ -116,18 +116,21 @@ def start_date_fields(item)
...
+    roles.flat_map do |role|
+      Array.wrap(role.strip
+        .gsub(/^Teacher$|Cover Support Teacher|TLRs|Lead Practitioner|Trainee Teacher|Peripatetic Music/, "teacher")
+        .gsub(/^Principal$|Head of School|Associate Principal|Executive Principal/, "headteacher")
+        .gsub("Vice Principal", "deputy_headteacher")
+        .gsub("Assistant Principal", "assistant_headteacher")
+        .gsub(/Head of Department|Head of Dept/, "head_of_department_or_curriculum")
+        .gsub(/Head of Year|Head of Phase/, "head_of_year_or_phase")
+        .gsub(/Teaching Assistant|Cover Support Teaching Assistant/, "teaching_assistant")
+        .gsub(%r{SEN/Inclusion Support|Pastoral|Technician|Librarian}, "education_support")
```
Comment (KyleMacPherson) [inline app/vacancy_sources/vacancy_source/source/ark.rb:131]: Could the the new roles such as `pastoral_health_and_welfare` for pastoral, or even just `other_support` for most of these be more appropriate?

File: app/vacancy_sources/vacancy_source/source/broadbean.rb [hunk 2]
```diff
@@ -80,19 +80,24 @@ def start_date_fields(item)
...
+      if role == "senior_leader"
+        Vacancy::SENIOR_LEADER_JOB_ROLES
+      elsif role == "middle_leader"
+        Vacancy::MIDDLE_LEADER_JOB_ROLES
+      else
+        Array.wrap(role.gsub("deputy_headteacher_principal", "deputy_headteacher")
+                       .gsub("assistant_headteacher_principal", "assistant_headteacher")
+                       .gsub("headteacher_principal", "headteacher")
+                       .gsub(/head_of_year_or_phase|head_of_year/, "head_of_year_or_phase")
+                       .gsub(/learning_support|other_support|science_technician/, "education_support")
```
Comment (KyleMacPherson) [inline app/vacancy_sources/vacancy_source/source/broadbean.rb:97]: We have a role called `other_support` now, so I guess we don't want to map that to `education_support`anymore

File: app/vacancy_sources/vacancy_source/source/every.rb [hunk 3]
```diff
@@ -103,19 +103,25 @@ def main_organisation(item)
...
+        Vacancy::SENIOR_LEADER_JOB_ROLES
+      elsif role == "middle_leader"
+        Vacancy::MIDDLE_LEADER_JOB_ROLES
+      else
+
+        Array.wrap(role.gsub("deputy_headteacher_principal", "deputy_headteacher")
+                       .gsub("assistant_headteacher_principal", "assistant_headteacher")
+                       .gsub("headteacher_principal", "headteacher")
+                       .gsub(/head_of_year_or_phase|head_of_year/, "head_of_year_or_phase")
+                       .gsub(/learning_support|other_support|science_technician/, "education_support")
```
Comment (KyleMacPherson) [inline app/vacancy_sources/vacancy_source/source/every.rb:121]: We have a role called `other_support` now, so I guess we don't want to map that to `education_support` anymore

File: app/vacancy_sources/vacancy_source/source/my_new_term.rb [hunk 4]
```diff
@@ -101,19 +101,25 @@ def find_schools(item)
...
+      if role == "senior_leader"
+        Vacancy::SENIOR_LEADER_JOB_ROLES
+      elsif role == "middle_leader"
+        Vacancy::MIDDLE_LEADER_JOB_ROLES
+      else
+        Array.wrap(role.gsub("deputy_headteacher_principal", "deputy_headteacher")
+                       .gsub("assistant_headteacher_principal", "assistant_headteacher")
+                       .gsub("headteacher_principal", "headteacher")
+                       .gsub(/head_of_year_or_phase|head_of_year/, "head_of_year_or_phase")
+                       .gsub(/learning_support|other_support|science_technician/, "education_support")
```
Comment (KyleMacPherson) [inline app/vacancy_sources/vacancy_source/source/my_new_term.rb:119]: We have a role called `other_support` now, so I guess we don't want to map that to `education_support`. Maybe `science_technician` would be better mapped to `other_support` too?

**Review by scruti — APPROVED** [review]
LGTM!

**Review by KyleMacPherson — APPROVED** [review]

**Pull request merged**